### PR TITLE
feat(minato): add `primary` type for auto generated primary key

### LINKED ISSUE #35: Feature: Introduce ObjectId as primary key for mongo
## Motivation
Auto-increment primary keys are introduced in cases where no fields can be used as primary naturally.
However, there are non-negligible overhead when using autoInc in mongo driver.
Therefore we propose to use the native ObjectId as auto-generated primary key for mongo and provide universal interface to support it while still maintaining consistency with other drivers.

## Design
Introduce new tagged type `Primary` (name is not decided yet) and new field type `primary`
The `Primary` type contains auto-generated value, and can only use as primary keys or foreign keys. Also it is unordered.
When `Primary` occurs as primary key, `autoInc` and `optimizeIndex` must be set.

In mongo driver, `Primary` is treated as ObjectId (therefore omitted in many situations), while in SQL-based drivers, `Primary` is treated as proper auto-increment intergers.

先占坑，看看有什么别的思路x

### PR DESCRIPTION
**Pull request by Hieuzest**
close #35

### CODE REVIEW

File: packages/core/src/model.ts [hunk 1]
```diff
@@ -22,13 +24,15 @@ export namespace Field {
...
   export const object: Type[] = ['list', 'json']
+  export const primary: Type[] = ['primary']
 
   export type Type<T = any> =
     | T extends number ? 'integer' | 'unsigned' | 'float' | 'double' | 'decimal'
     : T extends string ? 'char' | 'string' | 'text'
     : T extends boolean ? 'boolean'
     : T extends Date ? 'timestamp' | 'date' | 'time'
     : T extends unknown[] ? 'list' | 'json'
+    : T extends Primary ? 'primary' | 'unsigned'
```
Comment (shigma) [inline packages/core/src/model.ts:35]: why `unsigned` here?

Comment (Hieuzest) [reply packages/core/src/model.ts:35]: This allows mongo users to only change type `Primary` and still use old style autoInc as kind of compatibility.

File: packages/core/src/model.ts [hunk 2]
```diff
@@ -2,9 +2,9 @@ import { clone, isNullable, makeArray, MaybeArray } from 'cosmokit'
 import { Database } from './driver'
 import { Eval, isEvalExpr } from './eval'
 import { Selection } from './selection'
-import { Flatten, Keys } from './utils'
+import { Flatten, Indexable, Keys } from './utils'
```
Comment (shigma) [inline packages/core/src/model.ts:5]:
```suggestion
import { Flatten, Keys } from './utils'
```

File: packages/core/src/model.ts [hunk 3]
```diff
@@ -2,9 +2,9 @@ import { clone, isNullable, makeArray, MaybeArray } from 'cosmokit'
...
 
-export type Primary = (number | string) & { readonly _tag: unique symbol}
+export type Primary = Indexable & { [key: symbol]: true }
```
Comment (shigma) [inline packages/core/src/model.ts:7]:
```suggestion
export const Primary = Symbol('Primary')
export type Primary = (string | number) & { [Primary]: true }
```

**Review by shigma — APPROVED** [review]

**Pull request merged**